### ISSUE TITLE
Bloq counts fails for LessThanGate / SingleQubitCompare decomposition

### ISSUE THREAD

**Issue by fdmalone** [quantumlib/Qualtran#388]
This used to work with an older version of cirq_ft:

``` python
import cirq_ft
from qualtran.cirq_interop import CirqGateAsBloq

bloq = CirqGateAsBloq(cirq_ft.LessThanEqualGate(4, 4))
graph, sigma = get_bloq_counts_graph(bloq)
```

raises
```
BloqError: During finalization, {Soquet(binst=LeftDangle, reg=Register(name='less_than', bitsize=1, shape=(), side=<Side.THRU: 3>), idx=())} Soquets were not used.
```

Seems like the SingleQubitCompare is to blame

``` python
import cirq_ft
from qualtran.cirq_interop import CirqGateAsBloq

bloq = CirqGateAsBloq(cirq_ft.algos.SingleQubitCompare(False))
show_bloq(bloq.decompose_bloq())
```

```
BloqError: During finalization, {Soquet(binst=LeftDangle, reg=Register(name='less_than', bitsize=1, shape=(), side=<Side.THRU: 3>), idx=())} Soquets were not used.
```

**Comment by fdmalone** [comment]
@tanujkhattar any idea?

**Comment by tanujkhattar** [comment]
Looking into it now

**Comment by tanujkhattar** [comment]
It's because the target qubit in `And` gate in Cirq-FT now is a directional LEFT/RIGHT register, so it expects you to supply qubits which are newly allocated clean qubits; i.e. qubits which either appear in the signature of the parent gate as a LEFT / RIGHT register or don't appear in the signature of the parent gate at all and are allocated as `CleanQubit` / `BorrowableQubit`. 

Tl;Dr - The fix here is to change the signature of `SingleQubitCompare` to make `less_than` and `greater_than` directional registers; just like you would do if you write them as valid Bloqs. 

So this is less of a bug in the interop and more of a bug that arises due to old cirq-ft style implementation of the primitive where we didn't have directional registers. I'd imagine more such bugs would be discovered in other implementations and we can just fix them as we find them. 

I'll send a PR to fix this now.

**Comment by tanujkhattar** [comment]
@fdmalone https://github.com/quantumlib/Cirq/pull/6312 fixes the issue. It's hard to test the interop with the two codes living in separate repositories, but you can pull the Cirq PR, run `source dev_tools/pypath` in Cirq repository and then run your Qualtran code again to verify that this works. 

I'm working on moving all Cirq-FT code to Qualtran and we can then have automated consistency checks for all Cirq-FT gate's using `assert_valid_bloq_decomposition`.

**Comment by fdmalone** [comment]
I'm just reopening this so we remember to bump the cirq_ft requirements. I guess I can do that during #350.
